dali/pipeline/util/thread_pool.cc

```diff
@@ -57,24 +57,22 @@ ThreadPool::~ThreadPool() {
 #endif
 }
 
-void ThreadPool::AddWork(Work work, int64_t priority, bool finished_adding_work) {
-  std::lock_guard<std::mutex> lock(mutex_);
-  work_queue_.push({priority, std::move(work)});
-  work_complete_ = false;
-  adding_work_ = !finished_adding_work;
-}
-
-void ThreadPool::DoWorkWithID(Work work, int64_t priority) {
-  AddWork(std::move(work), priority, true);
-  // Signal a thread to complete the work
-  condition_.notify_one();
+void ThreadPool::AddWork(Work work, int64_t priority, bool start_immediately) {
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
+    work_queue_.push({priority, std::move(work)});
+    work_complete_ = false;
+    started_ |= start_immediately;
+  }
+  if (started_)
```

Comment:
perhaps we should do:
```
if (already started before) {
  condition_.notify_one();
} else if (starting because of this call) {
  condition_.notify_all();
}
```